util/nix/dirty-flake.nix

```diff
@@ -0,0 +1,35 @@
+{
+  description = "QMK firmware dirty flake expression";
+
+  # Load inputs from Niv and make them compatible with outputs args usage.
+  inputs = let
+    packageSources = import ./sources.nix {};
+  in {
+    flake-utils = import packageSources.flake-utils.outPath;
+    nixpkgs = packageSources.nixpkgs.outPath;
+    poetry2nix-src = packageSources.poetry2nix;
```

Comment:
I intended to make this section more generic, but didn't succeed. This manual fiddling made the outputs function get the correct attribute set.